internal/hns/hnsendpoint.go

```diff
@@ -222,6 +261,10 @@ func (endpoint *HNSEndpoint) HostDetach() error {
 	if err != nil {
 		return err
 	}
+	if windowsBuild == osversion.RS1 {
+		endpointMu.Lock()
+		defer endpointMu.Unlock()
+	}
 	return hnsCall("POST", "/endpoints/"+endpoint.Id+"/detach", string(jsonString), &response)
```

Comment:
Is concurrent `attach` / `detach` allowed? If so, these are not needed (thought I'd open this PR so that this approach can be discussed, and to make hcsshim the "source of truth" as to what is supported for each Windows version and what not, instead of putting that burden on consumers of this library 😅

Reply:
Not in RS1.

Reply:
so this looks ok (as in; yes: we should protect these endpoints as well), correct?

Would other endpoints need patching as well? (if they're not supported on RS1, they should probably produce an error?);

> On a side-note; wondering if os-version checks should be added to (e.g.) `HotAttachEndpoint()`, which (I think) is not supported on older versions

Reply:
@madhanrm Do we need to add version check around HotAttachEndpoint()?